services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -137,21 +138,34 @@ public class RouterHttpRequestStats extends AbstractVeniceHttpStats {
 
   // QPS metrics
   public RouterHttpRequestStats(
-      VeniceMetricsRepository metricsRepository,
+      MetricsRepository metricsRepository,
       String storeName,
       String clusterName,
       RequestType requestType,
       ScatterGatherStats scatterGatherStats,
       boolean isKeyValueProfilingEnabled) {
     super(metricsRepository, storeName, requestType);
-    emitOpenTelemetryMetrics = metricsRepository.getVeniceMetricsConfig().emitOtelMetrics();
-    openTelemetryMetricFormat = metricsRepository.getVeniceMetricsConfig().getMetricNamingFormat();
-    VeniceOpenTelemetryMetricsRepository otelRepository = metricsRepository.getOpenTelemetryMetricsRepository();
-    commonMetricDimensions = Attributes.builder()
-        .put(getDimensionName(VENICE_STORE_NAME), storeName)
-        .put(getDimensionName(VENICE_REQUEST_METHOD), requestType.name().toLowerCase())
-        .put(getDimensionName(VENICE_CLUSTER_NAME), clusterName)
-        .build();
+    VeniceOpenTelemetryMetricsRepository otelRepository;
+    if (metricsRepository instanceof VeniceMetricsRepository) {
+      VeniceMetricsRepository veniceMetricsRepository = (VeniceMetricsRepository) metricsRepository;
+      emitOpenTelemetryMetrics = (veniceMetricsRepository != null)
+          ? veniceMetricsRepository.getVeniceMetricsConfig().emitOtelMetrics()
+          : false;
+      openTelemetryMetricFormat = (veniceMetricsRepository != null)
+          ? veniceMetricsRepository.getVeniceMetricsConfig().getMetricNamingFormat()
+          : VeniceOpenTelemetryMetricNamingFormat.SNAKE_CASE;
```

Comment:
I could be wrong but this looks like the type of logic that will get repeated elsewhere once we start implementing more metrics... Can we have a static utility function in the `VeniceOpenTelemetryMetricNamingFormat` enum itself to centralize the determination of which enum value is the default one?

Reply:
After removing the `(veniceMetricsRepository != null)` as `instanceof` takes care it already, the default config is now part of the builder and not going to be repeated.